lib/std/os.zig

```diff
@@ -453,11 +454,54 @@ pub fn abort() noreturn {
         windows.kernel32.ExitProcess(3);
     }
     if (!builtin.link_libc and builtin.os.tag == .linux) {
+        // Linux man page wants to first "unblock SIG.ABRT", but this is a footgun
+        // for user-defined signal handlers that want to restore some state in
+        // some program sections and crash in others
+
+        // user installed SIGABRT handler is run, if installed
         raise(SIG.ABRT) catch {};
 
-        // TODO the rest of the implementation of abort() from musl libc here
+        // disable all signal handlers
+        sigprocmask(SIG.BLOCK, &linux.all_mask, null);
+
+        // TODO thread id should be taken from TLS instead of syscall
+        // TODO add thread group to TLS and use tgkill() instead of tkill()
+        // TODO conditionally add global atomic to start code and add CAS below:
+        //var abort_entered = std.atomic.Atomic(bool).init(false);
+        // ensure teardown by one thread
+        //while(abort_entered.compareAndSwap(false, true, .SeqCst, .SeqCst)) {}
+
+        // install default handler to terminate
+        const sigact = Sigaction{
+            .handler = .{ .sigaction = SIG.DFL },
+            .mask = undefined,
+            .flags = undefined,
+            .restorer = undefined,
+        };
+        sigaction(SIG.ABRT, &sigact, null) catch unreachable;
+
+        // make sure we have a pending SIGABRT queued
+        const tid = linux.gettid();
+        _ = linux.tkill(tid, SIG.ABRT);
+
+        // SIG.ABRT signal will run default handler
+        comptime var sigabrtmask_bitset = std.bit_set.ArrayBitSet(u32, @typeInfo(sigset_t).Array.len).initEmpty();
+        var sigabrtmask: system.sigset_t = undefined;
+        comptime sigabrtmask_bitset.set(SIG.ABRT);
+        inline for (sigabrtmask_bitset.masks) |mask, i|
+            sigabrtmask[i] = mask;
+        sigprocmask(SIG.UNBLOCK, &sigabrtmask, null);
```

Comment:
Avoid `comptime var` and `inline for`. Use constants for these things. Runtime bloat is especially tragic in this code which will almost never see the light of day.

Reply:
Should be fixed now. Calling `abort()` works at least without segfault/Kernel page fault, so the only active flag looks correct.